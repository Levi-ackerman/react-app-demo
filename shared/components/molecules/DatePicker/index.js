import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import RDatePicker from 'react-datepicker'
import moment from 'moment'
import 'react-datepicker/dist/react-datepicker.css'
import Icon from '../../atoms/Icon'

const Wrap = styled.div`
  border: 1px solid #e0e0e0;
  border-radius: 6px;
  height: 56px;
  padding: 0 16px;
  display: flex;
  align-items: center;
  background-color: #ffffff;
`

const Input = styled.input`
  color: #2d2d2d;
  font-size: 16px;
  margin-right: 16px;
  border: none;
  &:focus {
    outline: none;
  }
  ::placeholder {
    color: #939393;
    opacity: 1; /* Firefox */
  }

  :-ms-input-placeholder {
    /* Internet Explorer 10-11 */
    color: #939393;
  }

  ::-ms-input-placeholder {
    /* Microsoft Edge */
    color: #939393;
  }
`

class CustomInput extends React.PureComponent {
  render() {
    const { placeholder, value, onClick, onChange } = this.props
    return (
      <Wrap onClick={onClick}>
        <Input
          placeholder={placeholder}
          onChange={onChange}
          value={value}
          disabled
        />
        <Icon icon="down-arrow" />
      </Wrap>
    )
  }
}

class DatePicker extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      selected: props.selected ? props.selected : moment(),
    }
  }

  handleChange = date => {
    this.setState(
      {
        selected: date,
      },
      () => {
        if (this.props.onChange) {
          this.props.onChange(date)
        }
      },
    )
  }

  render() {
    return (
      <RDatePicker
        customInput={<CustomInput />}
        {...this.props}
        placeholderText="Select Date"
        dateFormat="DD MMM YYYY"
        selected={this.state.selected}
        onChange={this.handleChange}
      />
    )
  }
}

DatePicker.propTypes = {
  selected: PropTypes.object,
  onChange: PropTypes.func,
}

DatePicker.defaultProps = {
  selected: moment(),
}

export default DatePicker
