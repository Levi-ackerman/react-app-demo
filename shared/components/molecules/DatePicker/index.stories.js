import React from 'react'
import { storiesOf } from '@storybook/react'
import moment from 'moment'
import DatePicker from '.'

storiesOf('DatePicker', module).add('default', () => (
  <DatePicker selected={moment()} />
))
