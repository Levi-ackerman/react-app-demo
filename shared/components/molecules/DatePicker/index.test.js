import React from 'react'
import renderer from 'react-test-renderer'
import DatePicker from '.'
import moment from 'moment'

describe('DatePicker', () => {
  it('renders correctly', () => {
    const date = moment('2018-05-01 00:00:00')
    const tree = renderer.create(<DatePicker selected={date} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
