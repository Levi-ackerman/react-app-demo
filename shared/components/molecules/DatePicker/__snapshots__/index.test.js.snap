// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker renders correctly 1`] = `
<div>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <div
        className="DatePicker__Wrap-dBAPUL kKStES"
        onClick={[Function]}
      >
        <input
          className="DatePicker__Input-jzdVUe kdcKpb"
          disabled={true}
          onChange={[Function]}
          placeholder="Select Date"
          value="01 May 2018"
        />
        <span
          className="Icon__Wrapper-bdEMVZ kRGjUp"
          dangerouslySetInnerHTML={
            Object {
              "__html": "/asset/mock",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
